File: src/main/org/nlogo/window/InterfaceGlobalWidget.java
// (C) Uri Wilensky. https://github.com/NetLogo/NetLogo

package org.nlogo.window;

public interface InterfaceGlobalWidget {
  String name();

  String classDisplayName();

  Object valueObject();

  void valueObject(Object value);

  void updateConstraints();
}

File: netlogo-gui/src/main/window/InterfaceGlobalWidget.scala
// (C) Uri Wilensky. https://github.com/NetLogo/NetLogo

package org.nlogo.window

trait InterfaceGlobalWidget {
  def name: String
  def classDisplayName: String
  def valueObject(): AnyRef
  def valueObject(value: AnyRef): Unit
  def updateConstraints(): Unit
}
